src/utils/provider.ts

```diff
@@ -111,19 +111,17 @@ export class Block {
   tag: BlockIdentifier = null;
 
   private setIdentifier(__identifier: BlockIdentifier) {
-    if (typeof __identifier === 'string' && isHex(__identifier)) {
-      this.hash = __identifier;
-    } else if (typeof __identifier === 'bigint') {
+    if (isString(__identifier)) {
+      if (isHex(__identifier)) {
+        this.hash = __identifier;
+      } else if (validBlockTags.includes(__identifier as BlockTag)) {
+        this.tag = __identifier;
+      }
+    } else if (isBigInt(__identifier)) {
       this.hash = toHex(__identifier);
-    } else if (typeof __identifier === 'number') {
+    } else if (isNumber(__identifier)) {
```

Comment:
please check other instances in the codebase where this same logic could be applied 🙏